src/libraries/System.Private.CoreLib/src/System/Number.Formatting.cs

```diff
@@ -1683,14 +1699,31 @@ internal static unsafe void WriteTwoDigits(byte* ptr, uint value)
             return bufferEnd;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
I am not sure whether the split into aggressively inlined part is the right tradeoff here. It will make native code  callsites for methods like `uint.ToString()` a lot bigger. It will go from like ~7 bytes per callsite to ~50 bytes per callsite on x64, and even more with AOT.

The aggressive inlining helps with microbenchmarks. On the other hand, this is on a path that is assumed to allocate and so it is not likely to be super perf critical and small code size may be more important.

The aggressive inlining annotation is not the root cause of the problem. I expect everything would be still inlined even without explicit aggressive inlining annotation. The problem is that the logic is split into many small methods that the JIT will be always happy to inline. In other words, the split into small methods acts as implicit aggressive hint.